activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -30,6 +30,8 @@ def assert_not(object, message = nil)
       #   end
       def assert_nothing_raised
         yield
+      rescue => error
+        raise Minitest::UnexpectedError, error
```

Comment:
I actually wonder about the utility of adding this here, it seems minitest does it automatically: https://github.com/seattlerb/minitest/blob/15ed8e4ce504c8313058a1d6fc4918299be34328/lib/minitest/test.rb#L200-L201

Reply:
@kaspth that actually the whole point of the PR. If I take the PR description snippet again:

```ruby
    assert_raises RuntimeError do # rescue StandardError and compare with what is expected
      assert_no_difference -> { counter.count } do # made a noop by the RuntimeError
        counter.increment! # raises RuntimeError (subclass of StandardError)
      end
    end
```

So the idea here is to eagerly wrap the error in `Minitest::UnexpectedError` which doesn't inherit from `StandardError` so that `assert_raises` doesn't catch it and hide the legitimate issue.